tests/PolyType.Tests/AssociatedTypesTests.cs

```diff
@@ -1,19 +1,25 @@
 ﻿using PolyType;
 using PolyType.Tests;
+using Xunit.Internal;
 
-[assembly: TypeShapeExtension(typeof(AssociatedTypesTests.GenericDataType<,>), AssociatedTypes = [typeof(AssociatedTypesTests.GenericDataTypeVerifier<,>)])]
+[assembly: TypeShapeExtension(typeof(AssociatedTypesTests.GenericDataType<,>), AssociatedShapeDepth = TypeShapeDepth.Constructor, AssociatedTypes = [typeof(AssociatedTypesTests.GenericDataTypeVerifier<,>)])]
+[assembly: TypeShapeExtension(typeof(AssociatedTypesTests.GenericDataType<,>), AssociatedShapeDepth = TypeShapeDepth.All, AssociatedTypes = [typeof(AssociatedTypesTests.ExtraShape<,>)])]
```

Comment:
Do we have any tests checking that a depth is overriddent to `All` if the type is explicilty specified either directly via `GenerateShape` or transitively because a type containing it is generated?

Reply:
I thought so. But now I can't find it. I'll add one.